pkg/server/biz/scm/bitbucket/bitbucket-server/prs.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2017 caicloud authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// PullRequest represents a BitBucket Server pull request on a repository.
+type PullRequest struct {
+	ID          int    `json:"id"`
+	Version     int    `json:"version"`
+	Title       string `json:"title"`
+	Description string `json:"description"`
+	State       string `json:"state"`
+	Open        bool   `json:"open"`
+	Closed      bool   `json:"closed"`
+	CreatedDate int64  `json:"createdDate"`
+	UpdatedDate int64  `json:"updatedDate"`
+	FromRef     struct {
+		ID           string     `json:"id"`
+		DisplayID    string     `json:"displayId"`
+		LatestCommit string     `json:"latestCommit"`
+		Repository   Repository `json:"repository"`
+	} `json:"fromRef"`
+	ToRef struct {
+		ID           string     `json:"id"`
+		DisplayID    string     `json:"displayId"`
+		LatestCommit string     `json:"latestCommit"`
+		Repository   Repository `json:"repository"`
+	} `json:"toRef"`
+	Locked bool `json:"locked"`
+	Author struct {
+		User struct {
+			Name         string `json:"name"`
+			EmailAddress string `json:"emailAddress"`
+			ID           int    `json:"id"`
+			DisplayName  string `json:"displayName"`
+			Active       bool   `json:"active"`
+			Slug         string `json:"slug"`
+			Type         string `json:"type"`
+			Links        struct {
+				Self []struct {
+					Href string `json:"href"`
+				} `json:"self"`
+			} `json:"links"`
+		} `json:"user"`
+		Role     string `json:"role"`
+		Approved bool   `json:"approved"`
+		Status   string `json:"status"`
+	} `json:"author"`
+	Reviewers    []interface{} `json:"reviewers"`
+	Participants []interface{} `json:"participants"`
+	Links        struct {
+		Self []SelfLink `json:"self"`
+	} `json:"links"`
+}
+
+// PullRequestsService handles communication with the pull request related
+type PullRequestsService struct {
+	v1Client *V1Client
+}
+
+// GetPullRequest get a specific pull request.
+func (server *PullRequestsService) GetPullRequest(ctx context.Context, project string, repo string, number int) (*PullRequest, *http.Response, error) {
+	u := fmt.Sprintf("rest/api/1.0/projects/%s/repos/%s/pull-requests/%d", project, repo, number)
+	req, err := server.v1Client.NewRequest("GET", u, nil, nil)
```

Comment:
Use `http.MethodGet` instead of `GET`, please fix similar problems in other places.